.pre-commit-config.yaml

```
@@ -0,0 +1,13 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: debug-statements
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.3.5
+    hooks:
+      - id: ruff
+        args: [--config, "pyproject.toml", --fix]
+      - id: ruff-format
+        args: [--config, "pyproject.toml"]
```

See https://github.com/astral-sh/ruff-pre-commit?tab=readme-ov-file#using-ruff-with-pre-commit

```suggestion
    rev: v0.11.9
    hooks:
      - id: ruff-check
        args: [--config, "pyproject.toml", --fix]
      - id: ruff-format
        args: [--config, "pyproject.toml"]
```